pkg/metrics/metrics.go

```diff
@@ -126,27 +192,91 @@ func (opMgr *operationMetricsManager) RecordMetrics(op Operation, status Operati
 		opMgr.cache.Delete(op)
 		return
 	}
-	strStatus := opStatusUnknown
+	strStatus := string(SnapshotStatusTypeUnknown)
 	if status != nil {
 		strStatus = status.String()
 	}
-	duration := time.Since(ts).Seconds()
-	opMgr.opLatencyMetrics.WithLabelValues(op.Driver, op.Name, strStatus).Observe(duration)
+
+	// record invidual reconciliation time
+	reconciliationDuration := time.Since(ts).Seconds()
+	opMgr.opReconciliationLatencyMetrics.WithLabelValues(op.Driver, op.Name, op.SnapshotType, strStatus).Observe(reconciliationDuration)
```

Comment:
I am not sure this is the best place to implement reconciliation metrics.
the whole purpose of the cache is to reserve a starting point when there are multiple reconciliations. for a single reconciliation, we do not need any cached timestamp. If reconciliation latency is useful, then we should probably just do the duration calculation within controller logic. WDYT?